### ISSUE TITLE
Cookie values with = in them aren't handle properly

### ISSUE THREAD

**Issue by ShermanGirl** [MarkReedZ/mrhttp#10]
If you add the following test to requests it will fail with the value set to ar instead of b=ar

```python
  cookie = {'foo': 'b=ar'}
  r = requests.post('http://localhost:8080/printCookies', cookies=cookie)
  eq(r.text, "{'foo': 'b=ar'}")
```

The latest RFC is here http://www.ietf.org/rfc/rfc6265.txt - looks like some characters are marked invalid and mrhttp accepts them.  "US-ASCII characters excluding CTLs, whitespace DQUOTE, comma, semicolon, and backslash"

**Comment by MarkReedZ** [comment]
Good catch.  I'll add the test and fix it up after reviewing.  We probably should make sure we're following the RFC exactly for setting cookies in case a browser starts rejecting invalid chars.    

Looks like sanic doesn't follow the RFC so potentially someone could rely on comma being a valid character in the cookie then move from sanic to mrhttp and wonder whats going on if we drop the cookie.

**Comment by ShermanGirl** [comment]
https://bugs.python.org/issue19670

http cookies doesn't follow the RFC.  I tested it and the control characters plus comma and semicolon are octal escaped.  So whitespace, quotes, and backslash are output.  If a setter was added to request.cookies the cookie could be validated though backslash would have to be allowed...  

I say fix the =, don't allow control characters and leave the setting of the cookies as-is.  I'll play around with the cookie setting to make sure nothing surprising happens.
